### PR TITLE
Updated ROS2 PR: pcl_conversion only to ROS2 (dashing)

### PR DESCRIPTION
**Pull request by klintan**
@LucidOne I made another branch on my fork for (dashing compatible) which only contains the pcl_conversion part and not pcl_ros (COLCON_IGNORE added). 

Maybe this could be a first step to a ROS2 version, before the type masquerading etc is implemented in ROS2 ? 

Would love some feedback on this,

### CODE REVIEW

**Comment by klintan** [comment]
Perhaps someone can take a quick look at the updated travis config: It seems it's just picking up the old master-branch (aka ros1). Haven't used travis much so not sure where and how the code is fetched.

**Comment by mabelzhang** [comment]
There is currently a [version](https://github.com/ros2/pcl_conversions/tree/dashing) of `pcl_conversions` that is [released into Dashing](https://github.com/ros/rosdistro/blob/master/dashing/distribution.yaml#L1264) (Debian ros-dashing-pcl-conversions).

As mentioned here https://github.com/ros2/pcl_conversions/issues/5, the opinion from the creator @clalancette of that version is that, that version can be somehow merged with this PR and deprecated, before this repo can be released into Dashing. We can continue to port the rest of this repo (`pcl_ros`) to Dashing.

**Comment by mabelzhang** [comment]
I made a PR to merge the [ros2/pcl_conversions](https://github.com/ros2/pcl_conversions) that is officially released into Dashing but a bit dated, into this PR.
https://github.com/klintan/perception_pcl/pull/1

**Comment by klintan** [comment]
> I made a PR to merge the [ros2/pcl_conversions](https://github.com/ros2/pcl_conversions) that is officially released into Dashing but a bit dated, into this PR.
> [klintan#1](https://github.com/klintan/perception_pcl/pull/1)

Awesome thanks! I merged that one without any review so we can keep the review here (or wherever the final review takes place)

File: pcl_conversions/CMakeLists.txt
Comment (SteveMacenski) [inline pcl_conversions/CMakeLists.txt:31]: Please see this as a reference do doing includes / dependencies, this isn't easily maintainable 

https://github.com/ros-planning/navigation2/blob/master/nav2_planner/CMakeLists.txt

File: pcl_conversions/CMakeLists.txt
Comment (SteveMacenski) [inline pcl_conversions/CMakeLists.txt:45]: `${dependencies}` once you change how you do the includes

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [inline pcl_conversions/include/pcl_conversions/pcl_conversions.h]: You make a shared pointer to deference it, you just need the object. If for some reason you need a pointer, use the unique_ptr

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [inline pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Why did you change the namespace?

File: pcl_conversions/package.xml
Comment (SteveMacenski) [inline pcl_conversions/package.xml:8]: Adding maintainers?

File: perception_pcl/CMakeLists.txt
Comment (SteveMacenski) [inline perception_pcl/CMakeLists.txt]: new line at end of files

File: perception_pcl/package.xml
Comment (SteveMacenski) [inline perception_pcl/package.xml]: New line at end of files

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [inline pcl_conversions/include/pcl_conversions/pcl_conversions.h]: This is for type masquerading?

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
I'm going to assume all the time conversions with ns / us / s were done correctly or my brain will start hurting from looking at docs...

**Comment by SteveMacenski** [comment]
Another good question is where are the maintainers here? @wjwwood @wkentaro do you guys need some assistance to keep this going in ROS2?

File: pcl_conversions/package.xml
Comment (clalancette) [reply pcl_conversions/package.xml:8]: Author != maintainer .  Adding authors is generally fine, adding maintainers is generally a no-no :).

Comment (SteveMacenski) [reply pcl_conversions/package.xml:8]: Got it, I just looked at this and said "huh, why are you adding _other_ people unrelated?"

**Comment by clalancette** [comment]
I'll point out that CI is failing here, because of this message:

```
ERROR: the following packages/stacks could not have their rosdep keys resolved

to system dependencies:

pcl_conversions: Cannot locate rosdep definition for [pcl_msgs]

<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<

Function 'install_target_dependencies' returned with code '1' after 0 min 1 sec

The command ".industrial_ci/travis.sh" exited with 1.
```

It looks like someone needs to release `pcl_msgs` into dashing first.

**Comment by SteveMacenski** [comment]
@clalancette this is sounding a whole lot like my `robot_localization` ongoing effort. You just want to add me to `ros-perception` and I can hammer this all out? 

I dont like the idea of blocking process for `msgs` ;-) also I'm already maintaining image_pipeline and there's a bunch of PRs / issues in the SLAM projects I want to fix

**Comment by SteveMacenski** [comment]
So I'm trying to understand where things are at right now and I'm a bit confused. 

@mabelzhang you did some work to update @klintan branch, which as been merged and is represented in this PR, correct? 

@mabelzhang your PR homologates the ros2/pcl_conversions repository, correct?

So now if we merge this  (into what needs to be a new ros2/dashing branch, not crystal) and release it, we can nuke the ros2 fork (IMO not even archive it, delete it from existence since `pcl_conversions` `perception_pcl` were confusing enough in ROS1, now add in ROS2 some forks, some branches, etc....)

Is that a correct assessment?

Action item I can take
- [x] It doesnt look like there have been maintainers around in this for a while, I can try to get on that
- [x] PCL msgs aren't released, I can get on that
- [x] Make ros2/dashing branches
- [ ] Merge this PR
- [ ] Either begin porting pcl ros, which I probably can't commit to doing right at this moment, or make active inquires to parties I know have resources to do it

**Comment by clalancette** [comment]
> So now if we merge this (into what needs to be a new ros2/dashing branch, not crystal) and release it, we can nuke the ros2 fork (IMO not even archive it, delete it from existence since `pcl_conversions` `perception_pcl` were confusing enough in ROS1, now add in ROS2 some forks, some branches, etc....)

Just FYI, I'll do with the old `ros2/pcl_conversions` what I've done elsewhere, which is to do a "final" commit on the repository, removing all of the content and adding a README pointing to here.  Then I'll archive it.  At the very least we still need to keep it around for historical purposes and since we still support Crystal.

**Comment by SteveMacenski** [comment]
When's crystals cut off date?

Fair enough

**Comment by mabelzhang** [comment]
> I'm going to assume all the time conversions with ns / us / s were done correctly or my brain will start hurting from looking at docs...

I will leave that for @klintan to answer. The only changes I made to time were the `rclcpp::Time(uint sec, uint nanosec)` things in the test script, adding 0s to the calls to `ros::Time(double)` till there are 9 digits after the decimals for nanoseconds.

> It looks like someone needs to release pcl_msgs into dashing first.

That explains why I had to check out the `pcl_msgs` repo in order for colcon to find its headers.

> @mabelzhang you did some work to update @klintan branch, which as been merged and is represented in this PR, correct?

Yes.

> @mabelzhang your PR homologates the ros2/pcl_conversions repository, correct?

Yes.
I thought it might be hard to review the diffs since GitHub doesn't show diff between two different repos... So here's a list for reference. The only 6 differences between the two versions of `pcl_conversions.h` are as follows, and are resolved in my PRs:
1. Add `#include` for `<rclcpp/rclcpp.hpp>` and `message_filters` (present in perception_pcl PR, not present in pcl_conversions)
2. Use `rclcpp::Time` (in perception_pcl PR) instead of `builtin_interfaces::msg::Time` (in pcl_conversions). The latter used that type because `rclcpp::Time` didn't exist when that was ported. `melodic-devel` uses `ros::Time`, so the better replacement is `rclcpp::Time`.
3. In `PCLHeader <=> Header toPCL()`, set `seq` field to 0 (present in pcl_conversions, not present in perception_pcl PR)
4. Uncommented everything related to `pcl_msgs` (commented out in pcl_conversions, uncommented in perception_pcl PR)
5. Minor change in this comment (in perception_pcl PR), added `msg::` (in pcl_conversions):
`/** Provide to/fromROSMsg for sensor_msgs::PointCloud2 <=> pcl::PointCloud<T> **/`
6. Added commented code (in pcl_conversions) for the last block related to `message_traits`, because the convention for porting is to comment out ROS 1 code rather than to remove, so that when it's ready to port, there's an existing framework to base off of.

The test script, `package.xml`, and `CMakeLists.txt` were also merged.

> Either begin porting pcl ros, which I probably can't commit to doing right at this moment, or make active inquires to parties I know have resources to do it

We might be able to do this. That's my next thing to do.

> When's crystals cut off date?

Looks like Crystal EOL is Dec 2019 https://index.ros.org/doc/ros2/Releases/

**Comment by SteveMacenski** [comment]
@mabelzhang I'll take your word on the porting, everything I saw in this composite PR looks fine to me except for the things I pointed out, I just wanted to know if that was still a pending item on top of this PR or not. I assume you ran those tests and they came out clean?

With December 2019 being the cut off, and its essentially October 1 before this gets merged it, I'd even say we can just delete the `pcl_conversions` fork, but I don't really care that much to fight one way or another. Whatever's easiest.

**Comment by mabelzhang** [comment]
> I assume you ran those tests and they came out clean?

Yes, those tests in `test/` passed.

**Comment by klintan** [comment]
I've been super busy this week, but will fix all the outstanding issue on Saturday.

**Comment by mabelzhang** [comment]
I won't be able to get back to this till next week if there are things specific to my merge.

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (klintan) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: I think the templates for DefaultMessageCreator is in the message_filter namespace in ROS2; I get this error message trying to use the `ros` namespace 
```
-- stderr: pcl_conversions
In file included from /Users/andreasklintberg/personal/perception_pcl/pcl_conversions/test/test_pcl_conversions.cpp:5:
In file included from /Users/andreasklintberg/personal/perception_pcl/pcl_conversions/include/pcl_conversions/pcl_conversions.h:72:
In file included from /usr/local/include/pcl-1.9/pcl/io/pcd_io.h:791:
In file included from /usr/local/include/pcl-1.9/pcl/io/impl/pcd_io.hpp:49:
/usr/local/include/pcl-1.9/pcl/io/low_level_io.h:154:85: warning: missing field 'fst_bytesalloc' initializer [-Wmissing-field-initializers]
      ::fstore_t store = {F_ALLOCATEALL | F_ALLOCATECONTIG, F_PEOFPOSMODE, 0, length};
                                                                                    ^
In file included from /Users/andreasklintberg/personal/perception_pcl/pcl_conversions/test/test_pcl_conversions.cpp:5:
/Users/andreasklintberg/personal/perception_pcl/pcl_conversions/include/pcl_conversions/pcl_conversions.h:730:10: error: no template named 'DefaultMessageCreator'; did you mean 'message_filters::DefaultMessageCreator'?
  struct DefaultMessageCreator<pcl::PCLPointCloud2>
         ^~~~~~~~~~~~~~~~~~~~~
         message_filters::DefaultMessageCreator
/Users/andreasklintberg/personal/ros2/ros2-osx/include/message_filters/message_event.h:54:8: note: 'message_filters::DefaultMessageCreator' declared here
struct DefaultMessageCreator
       ^
In file included from /Users/andreasklintberg/personal/perception_pcl/pcl_conversions/test/test_pcl_conversions.cpp:5:
/Users/andreasklintberg/personal/perception_pcl/pcl_conversions/include/pcl_conversions/pcl_conversions.h:730:10: error: class template specialization of 'DefaultMessageCreator' not in a namespace enclosing 'message_filters'
  struct DefaultMessageCreator<pcl::PCLPointCloud2>
         ^
/Users/andreasklintberg/personal/ros2/ros2-osx/include/message_filters/message_event.h:54:8: note: explicitly specialized declaration is here
struct DefaultMessageCreator
       ^
In file included from /Users/andreasklintberg/personal/perception_pcl/pcl_conversions/test/test_pcl_conversions.cpp:5:
/Users/andreasklintberg/personal/perception_pcl/pcl_conversions/include/pcl_conversions/pcl_conversions.h:770:23: error: no template named 'HasHeader'; did you mean 'message_filters::message_traits::HasHeader'?
    template<> struct HasHeader<pcl::PCLPointCloud2> : std::true_type {};
                      ^~~~~~~~~
                      message_filters::message_traits::HasHeader
/Users/andreasklintberg/personal/ros2/ros2-osx/include/message_filters/message_traits.h:46:46: note: 'message_filters::message_traits::HasHeader' declared here
template<typename M, typename = void> struct HasHeader : public std::false_type {};
                                             ^
In file included from /Users/andreasklintberg/personal/perception_pcl/pcl_conversions/test/test_pcl_conversions.cpp:5:
/Users/andreasklintberg/personal/perception_pcl/pcl_conversions/include/pcl_conversions/pcl_conversions.h:770:23: error: class template specialization of 'HasHeader' not in a namespace enclosing 'message_traits'
    template<> struct HasHeader<pcl::PCLPointCloud2> : std::true_type {};
                      ^
/Users/andreasklintberg/personal/ros2/ros2-osx/include/message_filters/message_traits.h:46:46: note: explicitly specialized declaration is here
template<typename M, typename = void> struct HasHeader : public std::false_type {};
```

File: pcl_conversions/CMakeLists.txt
Comment (klintan) [reply pcl_conversions/CMakeLists.txt:31]: Hopefully I got it right, let me know if it looks ok :)

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (klintan) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: I was thinking just to make sure it gets freed when we go out of the scope. Is there another way without using `new` and `delete` ?

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (klintan) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Hm good question, I don't have an answer to that.

**Comment by klintan** [comment]
> I'm going to assume all the time conversions with ns / us / s were done correctly or my brain will start hurting from looking at docs...

Which ones are you referring to? 

I think the rest should be addressed except for the few follow up question I posed. (Namespace question and if everything looks OK in the CMakelists.txt file)

Let me know if there is still something outstanding on this.

**Comment by SteveMacenski** [comment]
Releasing PCL msgs to https://github.com/ros2-gbp/pcl_msgs-release

@clalancette @wjwwood  can you add me to the pcl_msgs repo so I can cut a new release on the ROS2 branch? Or just do it yourself if you have access

**Comment by mabelzhang** [comment]
@paulbovbel @wkentaro This PR is almost ready to go. The `pcl_conversions` fork officially released in Dashing is merged to here too. We want to get `perception_pcl` released in Eloquent, even if it requires putting `COLCON_IGNORE` in `pcl_ros`. We're just waiting on dependency `pcl_msgs` to be released. Could you take a look at this PR and add any comments regarding merging it?

**Comment by SteveMacenski** [comment]
PCL msgs are released now for dashing and eloquent. Still need a maintainer here to review this

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Why do you need a pointer at all? You just dereference it soon after

**Comment by SteveMacenski** [comment]
Created dashing  branch and retargeted PR to it, @klintan I see another draft PR, is that necessary?

I have write access to this repo now to help get this over  the line

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (klintan) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: No probably not. I was just thinking that if I use a smart pointer, it will automatically delete the object after the function scope is gone, no ? 

Constructing a new object would mean I need to do something like:

```
auto time_stamp = new rclcpp::Time(header.stamp);
fromPCL(pcl_header.stamp, time_stamp);
delete time_stamp;
```

But maybe that is not the way smart ptrs are intended to be used. Do you have any other recommendations on how to write this or is the code sample above an OK approach?

Comment (clalancette) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: You can just make a stack variable:

```
fromPCL(pcl_header.stamp, rclcpp::Time(header.stamp));
```

No smart pointers, pointers, or heap allocations required, and a single line of code :).

Comment (SteveMacenski) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: that's what I was getting at. There's no reason for this to be a pointer

File: pcl_conversions/CMakeLists.txt
Comment (SteveMacenski) [inline pcl_conversions/CMakeLists.txt]: I dont think you require this with the dependencies variable and the ament depends call

File: pcl_conversions/CMakeLists.txt
Comment (SteveMacenski) [inline pcl_conversions/CMakeLists.txt]: You didnt include any depends to the export depends `${dependencies}`

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [inline pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Why not use the commended out version? Also if that doesn't work, then remove from being commented out and delete it

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: if all this stuff is commented out, why leave the 2-3 things in it + namespace? This isn't a critique, I just don't know, is there something I dont know that needs that stuff?

File: pcl_conversions/package.xml
Comment (SteveMacenski) [inline pcl_conversions/package.xml]: This should be eigen3 I think

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
nice nice, we're super close!

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: @clalancette that's what this is for, correct?

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (klintan) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Ah ok awesome, I was (and I guess still am) a bit unsure on when it's possible to allocate on the stack and on the heap, but it seems after reading up on it more that it mostly comes down to the size of the object. If it's small enough always use the stack ( ?).

Thanks for your patience, I'll fix this.

Comment (SteveMacenski) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: There's a number of reasons to do one versus the other, but that is one reason, yes. 

There's a real performance downside with using heap memory so its best to not do it unless you require it. Plus if you look at what you actually ended up doing:

- create pointer
- dereference pointer to pass to a method

There wasn't anything that used it in pointer format, so there wasn't anything to be derived from the allocation

Comment (clalancette) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Typically, if you are only going to use it in this function (i.e. this stack frame), you do a stack allocation.  If it needs to live beyond this function, then you do a heap allocation (either with a "raw" pointer or a smart pointer).  Size can also come into it, as on Linux the default stack size is limited by default (8MB if I remember correctly, but don't quote me).

Comment (SteveMacenski) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Not to pile on, but also if you're concerned about real-time or software certifications, heap allocations are the devil

File: pcl_conversions/CMakeLists.txt
Comment (klintan) [reply pcl_conversions/CMakeLists.txt]: Had to include the non-ROS package here (PCL) otherwise I got an error, couldn't find the header.

Maybe I'm missing something ? since I had the same issue for the `navsat_transform_node` in the `robot_localization` package.

File: pcl_conversions/CMakeLists.txt
Comment (klintan) [reply pcl_conversions/CMakeLists.txt]: oops, fixed.

**Comment by klintan** [comment]
> Created dashing branch and retargeted PR to it, @klintan I see another draft PR, is that necessary?
> 
> I have write access to this repo now to help get this over the line

We could close that one, that one contains some changes to pcl_ros as well to try to convert to ROS2, but the effort stalled a bit (mostly because of the type masquerading not being available yet). My c++ skills are not on the level yet to be able to workaround that.

**Comment by SteveMacenski** [comment]
Got it, I guess just leave it there for now and I can close the draft PR when there's motion to getting pcl ros ported. An example still have alot of value

File: pcl_conversions/CMakeLists.txt
Comment (SteveMacenski) [inline pcl_conversions/CMakeLists.txt:23]: missing message filters in dependencies

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (SteveMacenski) [inline pcl_conversions/include/pcl_conversions/pcl_conversions.h:734]: Sorry maybe I was too non-specific, can you add ALL of this stuff back in, but then just comment it ALL out since there is no type masquerading yet? Revert to `namespace ros {...}` and just block comment it all out with a `TODO` on type masquerading.

File: pcl_conversions/package.xml
Comment (SteveMacenski) [inline pcl_conversions/package.xml:8]: Add me to the maintainer list if you would, so I can get build farm emails

File: pcl_conversions/package.xml
Comment (SteveMacenski) [inline pcl_conversions/package.xml]: Oops this should be `eigen` please change back :-)

Comment (SteveMacenski) [reply pcl_conversions/package.xml]: You can always check here https://github.com/ros/rosdistro/blob/master/rosdep/base.yaml

File: pcl_conversions/package.xml
Comment (SteveMacenski) [inline pcl_conversions/package.xml]: They're all exec depends, need to add them.

 if you want to simplify I'd support that instead. If you use the `<depend>` tag, it'll do  build, export, and execution dependency -- 1 liner to do all that above

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

**Comment by SteveMacenski** [comment]
Last few things to dot our 'I' and cross out 't'

File: pcl_conversions/include/pcl_conversions/pcl_conversions.h
Comment (klintan) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Thanks for all elaborate examples and explanations, really valuable!

Comment (klintan) [reply pcl_conversions/include/pcl_conversions/pcl_conversions.h]: Stack variable worked, however couldn't get it to compile using the one-liner 
`fromPCL(pcl_header.stamp, rclcpp::Time(header.stamp));`

**Comment by klintan** [comment]
> Last few things to dot our 'I' and cross out 't'

Yeah thanks for hanging in there, I think I got most of the kinks ironed out now, but if you find something new or if I missed something let me know.

**Comment by SteveMacenski** [comment]
This looks good to me. Let me look at this on Monday and play with on my machine. 

CI is failing, which seems related to switching to industrial ci. If you could make that pass that’d be great but I’ll also look into it

**Comment by klintan** [comment]
> This looks good to me. Let me look at this on Monday and play with on my machine.
> 
> CI is failing, which seems related to switching to industrial ci. If you could make that pass that’d be great but I’ll also look into it

Yeah I saw that
`pcl_conversions: No definition of [ament_cmake_gtest] for OS [debian]` 
failing on debian. 
Do you know if there is a docker for debian and dashing? Looking here https://github.com/osrf/docker_images/tree/1eb48f1275f0a9153d0b0ed088bed338f1a79ead/ros and docker hub I could only find ubuntu :( So I couldn't test it ( I mean I can always build my own, but just wanted to ask first if there might be one already out there, I'm guessing Travis is using one for instance)

**Comment by SteveMacenski** [comment]
I know alot of repos use the industrial CI so there must be a way to stop that. Otherwise everyone would be broken, those ament cmake packages are super low dependencies. For right now just targeting ubuntu is fine

**Comment by SteveMacenski** [comment]
@klintan just remove `     - OS_NAME=debian OS_CODE_NAME=stretch ` from your updated CI file. You don't need to run that

File: .travis.yml
Comment (SteveMacenski) [inline .travis.yml]: If you look at the github diff, you can see an example of how to do the matrix builds so we can easily test new distributions. Can you follow that styling? Eloquent is around the corner and there's a rosdistro file / docker containers so we should be able to do CI on Dashing and Eloquent.

**Review by SteveMacenski — CHANGES_REQUESTED** [review]
Just remove the matrix build for debian and make the matrix builds expandable, and we can merge this guy

File: .travis.yml
Comment (SteveMacenski) [reply .travis.yml]: Everything looks good on my machine so we just need to make CI happy and we can be finished!

Comment (klintan) [reply .travis.yml]: I change the global ros distro to be in the matrix instead, couldn't find any good example on how it's done in other places, hopefully that is what you meant :) 

edit: It passes now as well, since debian was removed.

**Review by SteveMacenski — APPROVED** [review]
@clalancette want to give this another once over? I'm happy with it

**Comment by clalancette** [comment]
> @clalancette want to give this another once over? I'm happy with it

Honestly, I'm not going to have time this week with ROSCon.  I've verified that cartographer_ros still builds with it, so that's going to have to be good enough.  Anything else we can fix in a follow-up.

**Pull request merged**
